src/cursor/cur_hs.c

```diff
@@ -146,6 +146,40 @@ __curhs_set_value_ptr(WT_CURSOR *hs_cursor, WT_CURSOR *file_cursor)
     F_SET(hs_cursor, F_MASK(file_cursor, WT_CURSTD_VALUE_SET));
 }
 
+/*
+ * __curhs_search --
+ *     Search the history store for a given key and position the cursor on it.
+ */
+static int
+__curhs_search(WT_CURSOR_BTREE *hs_cbt, WT_ITEM *srch_key, bool insert)
+{
+    WT_BTREE *hs_btree;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    hs_btree = CUR2BT(hs_cbt);
+    session = CUR2S(hs_cbt);
+
+#ifdef HAVE_DIAGNOSTIC
+    /*
+     * Turn off cursor-order checks in all cases on search. The search/search-near functions turn
+     * them back on after a successful search.
+     */
+    __wt_cursor_key_order_reset(hs_cbt);
+#endif
+
+    WT_ERR(__wt_cursor_func_init(hs_cbt, true));
+
+    WT_WITH_BTREE(
+      session, hs_btree, ret = __wt_row_search(hs_cbt, srch_key, insert, NULL, false, NULL));
+
```

Comment:
I think this needs:
```
WT_ERR(__wt_cursor_key_order_init(cbt));
```
after the search, otherwise we might not detect HS order failures?